### PR TITLE
podinfo: add a deleted pod cache

### PR DESCRIPTION
**Pull request by kkourt**
Add a deleted pod cache for pod association. See commits.

### CODE REVIEW

File: pkg/watcher/watcher_test.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+	*testing.T
+	Logger *logrus.Logger
+}
+
+func (tl tlog) Write(p []byte) (n int, err error) {
+	tl.Log(string(p))
+	return len(p), nil
+}
+
+func TestFastK8s(t *testing.T) {
```
Comment (mtardy) [inline pkg/watcher/watcher_test.go:39]: nit: it's not super explicit what the test does from its name, maybe we can add some comments here from the commit message or change the name?

**Review by mtardy — APPROVED** [review]
That's cool! :) lgtm

File: pkg/watcher/watcher_test.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+	informerFactory := informers.NewSharedInformerFactory(client, 0)
+	podInformer := informerFactory.Core().V1().Pods().Informer()
+	podInformer.AddEventHandler(ts.eventHandler())
+
+	watcher, err := newK8sWatcher(informerFactory)
+	require.Nil(t, err)
+	watcher.Start()
+
+	// Make sure informers are running.
+	//informerFactory.Start(ctx.Done())
```
[the remark this answers is not in the document]
Comment (kkourt) [reply pkg/watcher/watcher_test.go:80]: It's a leftover, removed.

File: pkg/watcher/watcher_test.go (hunk 1 above)
Comment (kkourt) [reply pkg/watcher/watcher_test.go:41]: Added a comment, thanks!

**Comment by kkourt** [comment]
Thanks for the review. Pushed a new version based on the feedback. Merging.

**Pull request merged**